pkg/secrethub/credentials/bootstrap_code.go

```diff
@@ -0,0 +1,155 @@
+package credentials
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/auth"
+	"github.com/secrethub/secrethub-go/internals/crypto"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/internals/http"
+)
+
+var _ Creator = (*BackupCodeCreator)(nil)
+var _ Provider = (*bootstrapCodeProvider)(nil)
+var _ auth.Signer = (*bootstrapCode)(nil)
```

Comment:
But they don't test. They enforce it. And they _show_ which part implements what interface.

Reply:
> But they don't test. They enforce it.

What's the difference? If we would run all tests in an `init` func, we would also enforce that they pass.
Maybe the difference is that this is verified at compile time rather than runtime?

> And they show which part implements what interface.

Yes this is an advantage of keeping them here! :+1:
Maybe add a short comment explaining it, I'm not sure whether it's clear to everyone at first glance.

Reply:
>  Maybe the difference is that this is verified at compile time rather than runtime?

That's what I meant.